javalanglib/src/main/scala/java/lang/Character.scala

```diff
@@ -519,6 +520,356 @@ object Character {
   @inline def compare(x: scala.Char, y: scala.Char): Int =
     x - y
 
+  // UnicodeBlock
+
+  class Subset protected (name: String) {
+    final override def equals(that: Any): scala.Boolean = super.equals(that)
+    final override def hashCode: scala.Int = super.hashCode
+    final override def toString(): String = name
+  }
+
+  final class UnicodeBlock private (name: String, private val start: Int, private val end: Int) extends Subset(name)
+
+  object UnicodeBlock {
+    private[this] val allBlocks: ArrayList[UnicodeBlock] = new ArrayList[UnicodeBlock](220)
+    private[this] val blocksByNormalizedName = new java.util.HashMap[String, UnicodeBlock]()
+
+    private[this] def addNameAliases(properName: String, block: UnicodeBlock): Unit = {
+      // Add normalized aliases
+      val lower = properName.toLowerCase
+      //   lowercase + spaces
+      blocksByNormalizedName.put(lower, block)
+      //   lowercase + no spaces
+      blocksByNormalizedName.put(lower.replaceAll("\\s", ""), block)
+      //   tokens as _
+      blocksByNormalizedName.put(lower.replaceAll("[\\s\\-_]", "_"), block)
```

Comment:
Let's try to get rid of the dependency on regexes, which are pretty heavy:

* `_` in the regex is redundant, since they need not be replaced.
* `\\s` could be replaced by the space `' '` AFAICT. It seems there never any other space characters than the actual space `' '`.
* If we only have the true space and `-` left, we can do `lower.replace(' ', '_').replace('-', '_')`

And if you're concerned about calling `replace` twice, you can store the result of replacing the spaces first, which is necessary two lines above anyway, and then only call the second `replace` on that one. That should be an overall win.